aksk/pkg/iam/driver/keystone/keystone.go

```diff
@@ -72,6 +76,26 @@ func (iam *KeystoneIam) CreateAkSk(aksk *model.AkSk, req *pb.AkSkCreateRequest)
 		panic(err)
 	}
 
+	// Validate UserId, TenantId .
+	if len(strings.TrimSpace(aksk.ProjectId)) == 0 || len(strings.TrimSpace(aksk.UserId)) == 0 {
```

Comment:
If you are validating strings input. It's better to validate by using regression. It will cover more special chars like -, ;, $, #, etc.

Reply:
validity of projectid , userid is checked with openstak api , here its just checking if its empty or not.

Reply:
Agree!!. But regression will check emptyness plus some other chars. Which helps to avoid unnecessary network call for checking at keystone level. Just my openion. otherwise ok!